src/gebt_poc/solver.cpp

```diff
@@ -557,4 +557,67 @@ Kokkos::View<double**> CalculateStaticIterationMatrix(
     return iteration_matrix;
 }
 
+Kokkos::View<double*> ConstraintsResidualVector(
+    const Kokkos::View<double*> gen_coords, const Kokkos::View<double*> position_vector
+) {
+    auto rotation_0 = gen_alpha_solver::EulerParameterToRotationMatrix(
+        gen_alpha_solver::create_vector({gen_coords(3), gen_coords(4), gen_coords(5), gen_coords(6)})
+    );
+    auto x0 =
+        gen_alpha_solver::create_vector({position_vector(0), position_vector(1), position_vector(2)}
+        );
+    auto u0 = gen_alpha_solver::create_vector({gen_coords(0), gen_coords(1), gen_coords(2)});
```

Comment:
Copy `u0` from `gen_coords` directly into `constraint_residual` since `u0` is a subview:

```c++
    // Copy u0 into constraint residual from gen coordinates
    Kokkos::deep_copy(u0, Kokkos::subview(gen_coords, Kokkos::make_pair(0, 3)));
```